src/mdns/mdns.cpp

```diff
@@ -486,6 +582,26 @@ void Publisher::HostRegistration::OnComplete(otbrError aError)
     }
 }
 
+bool Publisher::KeyRegistration::IsOutdated(const std::string &aName, const KeyData &aKeyData) const
+{
+    return !(mName == aName && mKeyData == aKeyData);
```

Comment:
I had to re-read the code to remember.

This is following  the same pattern as existing `IsOutDated()` methods for `HostRegisteration` and `ServiceRegisteration`. So I did the same pattern to be consistent. 

I agree none of them need to get a `aName` input and compare actually. It may make sense to change them all in separate PR to be consistent.